### ISSUE TITLE
$unflatten doesn't work

### ISSUE THREAD

**Issue by Curricane** [tafia/quick-xml#340]
I refer to the readme example to use $unflatten, but it doesn't work
```rust
/*
[package]
name = "quick_xml"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

# [dependencies]
# quick-xml = {version = "0.22.0", features = ["encoding","serialize"]}
# serde = {version = "1.0", feature = ["derive"]}
[dependencies]
serde = { version = "1.0", features = [ "derive" ] }
quick-xml = { version = "0.21", features = [ "serialize" ] }
*/
extern crate quick_xml;
extern crate serde;

// use quick_xml::de::{from_str, DeError};
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
struct Root {
    #[serde(rename = "$unflatten=first")]
    first: String,
    #[serde(rename = "$unflatten=second")]
    other_field: String,
}

fn test_root() {
    let s = "<root><first>value</first><second>value</second></root>";
    let d = quick_xml::de::from_str::<Root>(&s).unwrap();
    println!("d: {:?}", d); // error
}

fn test_root_ser() {
    let root = Root {
        first: "first".to_string(),
        other_field: "other_field".to_string(),
    };

    let s = quick_xml::se::to_string(&root).unwrap();
    println!("s: {}", s);
}
fn main() {
    // test_list_parts();
    // test_root(); //thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Custom("missing field `$unflatten=first`")',
    test_root_ser(); // s: <Root $unflatten=first="first" $unflatten=second="other_field"/>,
}
```

neither de nor se recognizes $unflatten.

I look forward to such an outcome：
```rust
<root><first>value</first><second>value</second></root>
```
however, quick-xml give me:
`<root first="value" second="value />`

**Comment by troffmo5** [comment]
unflattening was implemented after 0.22 release.
Try with:
`quick-xml = { version = "0.23.0-alpha3", features = ["serialize"] }`

**Comment by Tyilo** [comment]
This seems to work now.
